### PR TITLE
Ci/lighthouse workflow

### PR DESCRIPTION
**Pull request by nperon**
This PR adds a new workflow which consists of performance tests. 

The output is a set of indicators displayed in this very PR page. The "action page" link redirects to the action page where we find a "lighthouse-result" link to the lighthouse report zip file. This file, once unzipped, is the lighthouse report in the shape of a static site. The latter static site can be displayed locally in your navigator.

Limits formely set in file .github/lighthouse-budget.json are now released. Relevant documentation is at   [https://github.com/GoogleChrome/lighthouse/blob/master/docs/performance-budgets.md](https://github.com/GoogleChrome/lighthouse/blob/master/docs/performance-budgets.md)

### CODE REVIEW

**Comment by fredericvilcot** [comment]
@nperon You have a failed linting step in the CI (lint-yaml)

**Comment by nperon** [comment]
@fredericvilcot yaml linting is fixed. Thank you.

File: .github/workflows/lighthouse.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            const header = formatRow(['Performance','Accessibility','Best Practices','SEO','PWA','Action Page']);
+            const separator = formatRow(['---','---','---','---','---','---']);
+            const comment = [ header, separator, ...rows ].join('\n');
+            core.setOutput("comment", comment);
+
+      - name: Add Lighthouse stats as comment
+        id: comment_to_pr
+        uses: marocchino/sticky-pull-request-comment@39c5b5dc7717447d0cba270cd115037d32d28443 # v2.2.0
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (amimart) [inline .github/workflows/lighthouse.yml:92]:
```suggestion
          GITHUB_TOKEN: ${{ secrets.OKP4_TOKEN }}
```

Let github.com/bot-anik comment the PR.

**Review by amimart — CHANGES_REQUESTED** [review]
Thank you for contributing to the CI!

Well, I made some remarks but I think we need to discuss more general aspects of those changes.

First, I'm wondering if the chosen action is properly relevant, the workflow is commenting the PR with the populated indicators but this is done manually through scripting and I would have preferred the action to do that instead. The chosen action isn't capable of commenting the PR but I've seen some who can do it (e.g. https://github.com/foo-software/lighthouse-check-action). I don't know if you already considered this but, maybe we could consider changing the action used in order to simplify the workflow, what do you think ?

Another aspect I want to discuss are the commits messages, they are compliant with conventional commit, but some types may not be suitable. For instance, the `fix` commit type denote the solving of a functional issue, not a CI one. This is important because it is used to generate change logs and release notes, and a fix on the CI should not be presented as a functional fix.

File: .github/workflows/lighthouse.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        run: |
+          yarn build
+
+      - name: Build Storybook
+        run: |
+          yarn build-storybook
+
+      - name: Audit URLs using Lighthouse
+        id: lighthouse_audit
+        uses: treosh/lighthouse-ci-action@b4dfae3eb959c5226e2c5c6afd563d493188bfaf # 9.3.0
```
[the remark this answers is not in the document]
Comment (nperon) [reply .github/workflows/lighthouse.yml:29]: Back from some exploration on the other action suggested by @amimart namely [lighthouse-check-action](https://github.com/foo-software/lighthouse-check-action). The latter action cannot perform a audit on an artifact built and deployed within the github action context. Audit can be performed on sites whose urls are hardcoded in the workflow only.

[the remark this answers is not in the document]
Comment (nperon) [reply .github/workflows/lighthouse.yml]: Hash is now replaced with version

File: .github/workflows/lighthouse.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          LOCAL_SERVER_URL=`echo $GITHUB_CONTEXT | jq '.server_url'`
+          LOCAL_REPOSITORY=`echo $GITHUB_CONTEXT | jq '.repository'`
+          LOCAL_RUN_ID=`echo $GITHUB_CONTEXT | jq '.run_id'`
+          echo "action_server_url=$LOCAL_SERVER_URL" >> $GITHUB_ENV
+          echo "action_repository=$LOCAL_REPOSITORY" >> $GITHUB_ENV
+          echo "action_run_id=$LOCAL_RUN_ID" >> $GITHUB_ENV
+
+      - name: Format lighthouse score
+        id: format_lighthouse_score
+        uses: actions/github-script@7a5c598405937d486b0331594b5da2b14db670da # v6
```
[the remark this answers is not in the document]
Comment (nperon) [reply .github/workflows/lighthouse.yml]: Hash is now replaced with version

File: .github/workflows/lighthouse.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
+name: Lighthouse CI
+on: pull_request
```
[the remark this answers is not in the document]
Comment (nperon) [reply .github/workflows/lighthouse.yml]: Lighthouse flow is moved to lint.yml

File: .github/workflows/lighthouse.yml (hunk 1 above)
Comment (nperon) [reply .github/workflows/lighthouse.yml]: Cheange is made in the code

Comment (nperon) [reply .github/workflows/lighthouse.yml]: Correction for OKP4_TOKEN is now made everywhere.

**Review by amimart — CHANGES_REQUESTED** [review]
Hi, you have a merge commit on your history (i.e. `72241ecb9efeb9543ed19dbe82f3763a9fcaf055`) and we expect linear history, can you please make it in accordance?

**Comment by nperon** [comment]
This branch and the PR are discarded. Work is committed on another branch called ci/lighthouse_workflow_with_clean_branch

**Pull request closed without merging**